kratos/processes/vtk_output.cpp

```diff
@@ -0,0 +1,430 @@
+                rFileStream << "SCALARS " << element_result_name << " double"
+                           << " 1"
+                           << "\n";
+                rFileStream << "LOOKUP_TABLE default"
+                           << "\n";
+            }
+            else if (KratosComponents<Variable<array_1d<double, 3>>>::Has(element_result_name))
+            {
+                dataCharacteristic =  VtkOutput::WriteDataType::VTK_VECTOR;
+                rFileStream << "VECTORS " << element_result_name << " double"
+                           << "\n";
+            }
+
+            // write nodal results
+            for (const auto& elem : rModelPart.Elements())
+            {
+                if (dataCharacteristic ==  VtkOutput::WriteDataType::VTK_SCALAR)
+                {
+                    Variable<double> element_result_variable = KratosComponents<Variable<double>>::Get(element_result_name);
+                    const auto &element_result = elem.GetValue(element_result_variable);
+                    WriteScalarDataToFile(element_result, rFileStream);
+                    rFileStream <<"\n";
+                }
+                else if (dataCharacteristic == VtkOutput::WriteDataType::VTK_VECTOR)
+                {
+                    Variable<array_1d<double, 3>> element_result_variable = KratosComponents<Variable<array_1d<double, 3>>>::Get(element_result_name);
+                    const auto &element_result = elem.GetValue(element_result_variable);
+                    WriteVectorDataToFile(element_result, rFileStream);
+                    rFileStream << "\n";
+                }
+            }
+        }
+
+    }
+}
+
+void VtkOutput::WriteModelPart(const ModelPart &rModelPart)
+{
+    Initialize(rModelPart);
+
+    // Make the file stream object
+    std::string output_file_name = GetOutputFileName(rModelPart);
+    std::ofstream output_file;
+    if (mFileFormat == VtkOutput::FileFormat::VTK_ASCII)
+    {
+        output_file.open(output_file_name, std::ios::out | std::ios::trunc);
+        output_file << std::scientific;
+        output_file << std::setprecision(mDefaultPrecision);
+    } else if (mFileFormat == VtkOutput::FileFormat::VTK_BINARY)
+    {
+        output_file.open(output_file_name, std::ios::out | std::ios::binary | std::ios::trunc);
+    }
+
+    WriteHeader(rModelPart, output_file);
+    WriteMesh(rModelPart, output_file);
+    //WriteNodalResultsAsPointData(rModelPart, output_file);
+    //WriteElementData(rModelPart, output_file);
+
+    output_file.close();
+}
+
+void VtkOutput::PrintOutput()
+{
+    //For whole model part
+    WriteModelPart(mrModelPart);
+    //For sub model parts
+    bool print_sub_model_parts = this->mrOutputSettings["output_sub_model_parts"].GetBool();
+
+    if(print_sub_model_parts)
+    {
+        for (const auto& sub_model_part : mrModelPart.SubModelParts())
+        {
+            std::string output_file_name = GetOutputFileName(sub_model_part);
```

Comment:
maybe you could add sth to the file-name such that one can identify the level

e.g. submodelparts would get 1_Name
subsubmodelparts would get 11_name or 2_name

Reply:
we can discuss it tomorrow !